### PR TITLE
[go-server] Add ability to handle nullable query param

### PR DESCRIPTION
**Pull request by lwj5**
Add the ability to handle nullable query param

nullable query params will now be passed as a pointer to the servicer.
non-nullable query params will still be passed as value.

There is no breaking changes in this PR

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/go-server/service.mustache [hunk 1]
```diff
@@ -23,7 +23,7 @@ func New{{classname}}Service() {{classname}}Servicer {
 {{#isDeprecated}}
 // Deprecated
 {{/isDeprecated}}
-func (s *{{classname}}Service) {{nickname}}(ctx context.Context{{#allParams}}, {{paramName}} {{dataType}}{{/allParams}}) (ImplResponse, error) {
+func (s *{{classname}}Service) {{nickname}}(ctx context.Context{{#allParams}}, {{paramName}} {{#isNullable}}*{{/isNullable}}{{dataType}}{{/allParams}}) (ImplResponse, error) {
```
Comment (icubbon) [inline modules/openapi-generator/src/main/resources/go-server/service.mustache:26]: I don't `isNullable` is behaving as you expect it to here. The Pet Store Specs have many non-required params that would show this change but none of them are showing up in this diff.

Comment (lwj5) [reply modules/openapi-generator/src/main/resources/go-server/service.mustache:26]: could you provide a specific example? are they marked as nullable?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java [hunk 2]
```diff
@@ -322,34 +324,79 @@ public ModelsMap postProcessModels(ModelsMap objs) {
...
+                iterator.remove();
+        }
 
         boolean addedTimeImport = false;
         boolean addedOSImport = false;
+        boolean addedReflectImport = false;
         for (CodegenOperation operation : operations) {
             for (CodegenParameter param : operation.allParams) {
                 // import "os" if the operation uses files
-                if (!addedOSImport && ("*os.File".equals(param.dataType) || ("[]*os.File".equals(param.dataType)))) {
```
Comment (icubbon) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java:341]: Why this change? Would this break if an endpoint takes in a list/slice/array of Files?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java [hunk 3]
```diff
@@ -322,34 +324,79 @@ public ModelsMap postProcessModels(ModelsMap objs) {
...
+        }
+
+        // remove model imports to avoid error
         List<Map<String, String>> imports = objs.getImports();
         if (imports == null)
             return objs;
 
-        // override imports to only include packages for interface parameters
-        imports.clear();
+        System.out.println(imports);
```
Comment (icubbon) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java:341]: Debug

File: modules/openapi-generator/src/main/resources/go-server/service.mustache (hunk 1 above)
Comment (icubbon) [reply modules/openapi-generator/src/main/resources/go-server/service.mustache:26]: One specific example from `openapi-generator/samples/server/petstore/go-api-server/api/openapi.yaml` is the `booleanTest` query parameter in `deleteUser`.

```
- description: boolean query parameter
        explode: true
        in: query
        name: boolean_test
        required: false
        schema:
          type: boolean
        style: form
```

This should result in the the function signature for `deleteUser` having a `*bool` parameter for the function signature in `openapi-generator/samples/server/petstore/go-api-server/go/api_user_service.go`

There are numerous query parameters in that example file that have optional query parameters.

**Comment by lwj5** [comment]
To fix other import issues with os workaround in the future. ref: https://github.com/OpenAPITools/openapi-generator/issues/11097

File: modules/openapi-generator/src/main/resources/go-server/service.mustache (hunk 1 above)
Comment (lwj5) [reply modules/openapi-generator/src/main/resources/go-server/service.mustache:26]: object is not nullable, please see https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#schema-object

**Comment by lwj5** [comment]
Please be patient, I'm still fixing test issues

**Comment by icubbon** [comment]
I'm still researching and reading about how `nullable` is handled in other generators.

What is currently giving me hesitation is comparing this implementation to how the `go-client` handles non-required parameters. Below is how the go-client determines if a parameter will be a pointer or a value:
`	{{name}} {{^required}}{{^isNullable}}{{^isArray}}{{^isFreeFormObject}}*{{/isFreeFormObject}}{{/isArray}}{{/isNullable}}{{/required}}{{{dataType}}} `json:"{{{baseName}}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{{baseName}}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
`
This specific line is from `modules/openapi-generator/src/main/resources/go/model_simple.mustache`

**Review by icubbon — COMMENTED** [review]
Examples should be added/edited to cover the new variations of nullable bools, nullable enums, nullable numbers, etc.

File: samples/openapi3/server/petstore/go/go-petstore/go/api_pet.go [hunk 4]
```diff
@@ -270,10 +271,15 @@ func (c *PetAPIController) UploadFile(w http.ResponseWriter, r *http.Request) {
...
+	var fileParam *os.File
+	{
+		param, err := ReadFormFileToTempFile(r, "file")
+		if err != nil {
+			c.errorHandler(w, r, &ParsingError{Err: err}, nil)
+			return
+		}
+
+		fileParam = param
 	}
```
[the remark this answers is not in the document]
Comment (lwj5) [reply samples/openapi3/server/petstore/go/go-petstore/go/api_pet.go:283]: I understand that this is uncommon, but this is purely cosmetic and does not affect the semantics of the code. 

This will need to be fixed with the larger problem as stated above https://github.com/OpenAPITools/openapi-generator/pull/17051#issuecomment-1814783198

**Comment by lwj5** [comment]
> #11097
